contracts/non-fungible-tokens/andromeda-crowdfund/src/testing/tests.rs

```diff
@@ -587,6 +662,398 @@ mod test {
+                expected_res: Err(ContractError::InsufficientFunds {}),
+                payee: buyer.to_string(),
+                start_time: Some(past_time()),
+                end_time: future_time(&env),
+                orders: vec![SimpleTierOrder {
+                    level: Uint64::one(),
+                    amount: Uint128::new(20),
+                }],
+                initial_cap: Uint128::new(500),
+                funds: vec![coin(10, MOCK_NATIVE_DENOM)],
+                denom: Asset::NativeToken(MOCK_NATIVE_DENOM.to_string()),
+            },
+            PurchaseTierTestCase {
+                name: "Purchase in wrong campaign stage using native tokens".to_string(),
+                stage: CampaignStage::READY,
+                expected_res: Err(ContractError::InvalidCampaignOperation {
+                    operation: "purchase_tiers".to_string(),
+                    stage: "READY".to_string(),
+                }),
+                payee: buyer.to_string(),
+                start_time: Some(past_time()),
+                end_time: future_time(&env),
+                orders: vec![SimpleTierOrder {
+                    level: Uint64::one(),
+                    amount: Uint128::new(10),
+                }],
+                initial_cap: Uint128::new(500),
+                funds: vec![coin(1000, MOCK_NATIVE_DENOM)],
+                denom: Asset::NativeToken(MOCK_NATIVE_DENOM.to_string()),
+            },
+            PurchaseTierTestCase {
+                name: "Purchase before campaign start using native tokens".to_string(),
+                stage: CampaignStage::ONGOING,
+                expected_res: Err(ContractError::CampaignNotStarted {}),
+                payee: buyer.to_string(),
+                start_time: Some(future_time(&env)),
+                end_time: future_time(&env),
+                orders: vec![SimpleTierOrder {
+                    level: Uint64::one(),
+                    amount: Uint128::new(10),
+                }],
+                initial_cap: Uint128::new(500),
+                funds: vec![coin(1000, MOCK_NATIVE_DENOM)],
+                denom: Asset::NativeToken(MOCK_NATIVE_DENOM.to_string()),
+            },
+            PurchaseTierTestCase {
+                name: "Purchase after campaign end using native tokens".to_string(),
+                stage: CampaignStage::ONGOING,
+                expected_res: Err(ContractError::CampaignEnded {}),
+                payee: buyer.to_string(),
+                start_time: Some(past_time()),
+                end_time: past_time(),
+                orders: vec![SimpleTierOrder {
+                    level: Uint64::one(),
+                    amount: Uint128::new(10),
+                }],
+                initial_cap: Uint128::new(500),
+                funds: vec![coin(1000, MOCK_NATIVE_DENOM)],
+                denom: Asset::NativeToken(MOCK_NATIVE_DENOM.to_string()),
+            },
+            PurchaseTierTestCase {
```

Comment:
Should we add a test case for purchasing more than the maximum amount in the tier?

Reply:
Added test case for purchasing more than limit using native token